x-pack/plugins/security/public/account_management/api_keys_page/invalidate_api_key_modal.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FunctionComponent } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiText } from '@elastic/eui';
+import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+import { ApiKeyToInvalidate } from '../../../common/model';
+import {
+  APIKeysAPIClient,
+  InvalidateApiKeysResponse,
+} from '../../management/api_keys/api_keys_api_client';
+import { useSubmitHandler } from '../components/use_form';
+import { ConfirmModal, ConfirmModalProps } from '../components/confirm_modal';
+
+export interface InvalidateApiKeyModalProps {
+  onCancel: ConfirmModalProps['onCancel'];
+  onSuccess?(result: InvalidateApiKeysResponse): any;
+  onError?(error: Error): any;
+  apiKey: ApiKeyToInvalidate;
+}
+
+export const InvalidateApiKeyModal: FunctionComponent<InvalidateApiKeyModalProps> = ({
+  onCancel,
+  onSuccess,
+  onError,
+  apiKey,
+}) => {
+  const { services, notifications } = useKibana();
+  const [state, invalidateApiKeys] = useSubmitHandler(
+    {
+      onSubmit: () =>
+        new APIKeysAPIClient(services.http!).invalidateApiKeys([
+          { id: apiKey.id, name: apiKey.name },
+        ]),
+      onSubmitSuccess: (values) => {
+        notifications.toasts.success({
+          iconType: 'check',
+          title: i18n.translate(
+            'xpack.security.accountManagement.invalidateApiKey.successMessage',
+            {
+              defaultMessage: 'Invalidated API key “{name}”',
+              values: { name: apiKey.name },
+            }
+          ),
+          toastLifeTimeMs: 3000,
```

Comment:
**question** any reason not to use the standard timeout duration for this toast?

Reply:
There was a bug in the notifications service that defaulted this to undefined / 0. 

I have not fixed that bug instead of manually setting the timeout duration.